src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -113,27 +107,29 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                     return;
                 }
 
-                ArrayBuilder<BoundStatement>? block = printableMembers.IsEmpty ? null : ArrayBuilder<BoundStatement>.GetInstance();
+                ArrayBuilder<BoundStatement> block;
                 BoundParameter builder = F.Parameter(this.Parameters[0]);
-                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType())
+                if (ContainingType.BaseTypeNoUseSiteDiagnostics.IsObjectType() || ContainingType.IsRecordStruct)
                 {
                     if (printableMembers.IsEmpty)
                     {
                         // return false;
                         F.CloseMethod(F.Return(F.Literal(false)));
                         return;
                     }
+                    block = ArrayBuilder<BoundStatement>.GetInstance();
```

Comment:
>block = ArrayBuilder<BoundStatement>.GetInstance(); [](start = 20, length = 51)

Consider moving `block`s declaration and initialization after the entire `if`.

Reply:
I'll leave this as-is. Seems fine

---
In reply to: [601939283](https://github.com/dotnet/roslyn/pull/52012#discussion_r601939283) [](ancestors = 601939283)